### PR TITLE
add multi support mentions

### PR DESCRIPTION
**Pull request by fancykingstar**
- Revise mention package to support multi mentions.

### CODE REVIEW

**Comment by vas85** [comment]
@sibelius this is the same PR as the earlier one here: https://github.com/draft-js-plugins/draft-js-plugins/pull/1419

@fancykingstar and I work together. It would be great if we can get this merged

File: .eslintrc.js [hunk 1]
```diff
@@ -86,7 +86,7 @@ module.exports = {
     'jsx-a11y/img-has-alt': 0,
     'react/require-default-props': 0,
     'react/display-name': 1,
-
+    'linebreak-style': 0,
```
Comment (fxOne) [inline .eslintrc.js:89]: Why did you disable this rule?

**Comment by fancykingstar** [comment]
without that I got the error like above in the windows.
Expected linebreaks to be 'LF' but found 'CRLF' linebreak-style

And that rule helps me to avoid it.

**Comment by fancykingstar** [comment]
Hi @sibelius 
Would you review the PR and merge?

**Comment by sibelius** [comment]
can you add this to storybook and docs as well?

**Comment by fancykingstar** [comment]
I think I've already added.

**Comment by fxOne** [comment]
> without that I got the error like above in the windows. Expected linebreaks to be 'LF' but found 'CRLF' linebreak-style And that rule helps me to avoid it.

The idea of the rule is that the code has only one type linebreaks and not mixed ones. Can you please enable the rule and fix potential issues.

File: packages/mention/src/index.tsx [hunk 2]
```diff
@@ -58,8 +59,10 @@ export interface MentionPluginConfig {
   mentionSuggestionsComponent?: ComponentType;
   entityMutability?: 'SEGMENTED' | 'IMMUTABLE' | 'MUTABLE';
   mentionTrigger?: string;
+  mentionTriggers?: string[];
```
Comment (fxOne) [inline packages/mention/src/index.tsx:62]: If we have now `mentionTrigger` and `mentionTriggers` what happens if you set both?

**Review by fxOne — CHANGES_REQUESTED** [review]

**Comment by fxOne** [comment]
> I think I've already added.

You missed at least https://github.com/draft-js-plugins/draft-js-plugins/blob/master/packages/docs/pages/plugin/mention/index.tsx and the changelog.

File: packages/mention/src/index.tsx (hunk 2 above)
Comment (fancykingstar) [reply packages/mention/src/index.tsx:62]: - If you'd like to use only one trigger, then you should use mentionTrigger like above
`const mentionPlugin = createMentionPlugin({ mentionTrigger: '(' });`

- if you'd like to use multiple triggers, then you should use mentionTriggers like
`const mentionPlugin = createMentionPlugin({ mentionTriggers: ['@', '#'] });`

I think this makes more sense.

**Comment by fancykingstar** [comment]
@sibelius 
Would you take a look at PR and do me a favor to handle the error.

**Comment by fancykingstar** [comment]
@sibelius Actually, I don't have this kind of error in the local.
I run 'npm run lint' to test lint in the local.

**Comment by fancykingstar** [comment]
image
@sibelius @fxOne 
I just did strict clone and run "yarn lint" in the local machine.
But I got the error like above.

**Comment by fxOne** [comment]
Run a `yarn build` to build the dependencies.

**Comment by fancykingstar** [comment]
image
I got this error when I run "yarn build" @fxOne

**Comment by fancykingstar** [comment]
image

**Comment by fxOne** [comment]
You need to run it under linux or go into each folder and run yarn build except the docs folder

**Comment by fancykingstar** [comment]
I run it under mac. doesn't it work?

**Comment by fancykingstar** [comment]
I use yarn 1.22.4

**Comment by fancykingstar** [comment]
image
I run "yarn build" in each subfolder and I got the error.

**Comment by fxOne** [comment]
I just tried it under Windows with WSL2 and ubuntu 20.4 where it worked:
build
I have never used a mac so I cannot help you

**Comment by fancykingstar** [comment]
Hi @sibelius @fxOne Would you review the PR and give me feedback?

**Comment by fancykingstar** [comment]
@sibelius When could you review the PR.

**Comment by sibelius** [comment]
I'm gonna have some time next week

**Comment by sibelius** [comment]
I'm gonna have some time next week

**Comment by fxOne** [comment]
I just tried the PR an notice that most of the mention storybook do not work (Mention Suggestions are not showing). Can you please check this?

**Comment by fancykingstar** [comment]
image
Storybook works fine. Please checkout this image.

**Comment by fxOne** [comment]
I tried it again and still not working:
Unbenannt

**Comment by fancykingstar** [comment]
image
Sorry for confusion. I revised it and will push the updates.

**Comment by fancykingstar** [comment]
@fxOne Would you review again?

**Comment by fxOne** [comment]
The storybook examples `Editor with Mention Plugin and asynchronously loaded suggestions` and `Mentions with whitespace` are still not working for me.

**Comment by fancykingstar** [comment]
Hi @fxOne.
I tried the storybook of master branch of draft-js-plugin and it was not working either. That means there isn't issue in my PR.
And I tested all the other storybook, docs, and examples, and they works as same as the original draft-js-plugin.

FYI, storybook for `Editor with Mention Plugin and asynchronously loaded suggestions` is not working because of base PATH.

**Comment by fxOne** [comment]
I see the issue with `Editor with Mention Plugin and asynchronously loaded suggestions`. I'll try to fix it.

**Comment by vas85** [comment]
@fxOne / @sibelius can we get a review of this please?

This feature has been asked for repeatedly for 6+ months by many people. It would be great to get this merged so that all users of this package can benefit.

**Comment by sibelius** [comment]
We can't merge code that is not ready

**Comment by vas85** [comment]
1. The issue you brought up already exists in this project and has nothing to do with this PR.
2. Other issues you brought up have been addressed

Please review and add comments specific to this PR. Existing issues are outside the scope of this PR

**Comment by fxOne** [comment]
I still there an issue with the `Mentions with whitespace` example which works for on master but not in the PR

File: packages/mention/src/index.tsx (hunk 2 above)
Comment (fxOne) [reply packages/mention/src/index.tsx:66]: can you change it to `mentionTrigger?: string | string[];` please?

**Pull request closed without merging**